queries/shellscript.scm

```diff
@@ -0,0 +1,161 @@
+;;
+;; Statements
+;;
+
+(program
+    (_) @statement
+)
+[
+    (if_statement)
+    (command)
+    (function_definition)
+    (variable_assignment)
+] @statement
+
+;;
+;; Conditionals
+;;
+
+;;!!
+(if_statement
+    (_)*
+    "then"
+    (_)*
+    "fi"
+) @ifStatement @branch
+
+(if_statement
+    "if" @branch.start.startOf
+    (_)
+    "then"
+    (_)* @branch.interior
+    .
+    [
+        (elif_clause)
+        (else_clause)
+    ] @branch.end.startOf
+) @ifStatement @branch.iteration @_.domain
+
+(elif_clause
+    (_)* @condition
+    "then"
+    (_)* @branch.interior
+) @branch
+
+(else_clause
+    "else" @branch.interior.start.endOf
+    (_) @branch.interior.end.endOf
+) @branch
+
+(_
+    condition: (_) @condition
+) @_.domain
+
+;;
+;; Lists and maps
+;;
+
+;;!! array=("a" "b" "c")
+;;!        ^^^^^^^^^^^^^
+;;!        -------------
+(array
+    "(" @_.interior.start.endOf
+    (_)? @collectionItem
+    ")" @_.interior.end.startOf
+) @list
+
+;;
+;; Strings
+;;
+
+;;!! # foo
+;;!  ^^^^^
+(comment) @comment @textFragment
+
+;;!! var="foo"
+;;!      ^^^^^
+(string) @string @textFragment @argumentOrParameter.iteration
```

Comment:
There was a discussion that some of you had in another issue which I forget off hand, where iirc the consensus was that ${} expansions inside of strings were going to be referred to be as arg scopes. So I've been testing that on the shell script stuff and it seems to work pretty nicely. I can't remember offhand if I added it to nix as well, which uses similar expansions.

Reply:
ohh right yeah if interpolations are args, then string should be iteration scope. Feels a bit surprising to get trapped in a string for "every arg" tho. cc/ @AndreasArvidsson

Reply:
Yeah I'm a bit hesitant to make strings the iteration scope for arguments.

Reply:
So there are at least 3 options I guess?

1) Only add an iteration scope if the string actually has an interpolation inside, so it's less surprising 
2) Keep interpolation as args, but just don't have an iterarion scope for the container strings (I feel like being able to target every interpolation in a string could be nice though?)
3) Create a new scope and don't use args

Reply:
1) is an interesting possibility. We don't usually restrict depending on whether the scope actually appears, but I could see an argument here

Re 2), you don't actually need to have an iteration scope to target every interpolation in a string. You'd just say `"every arg string"`. Iteration scope is only used when there's no explicit range given for iteration. That being said, in https://github.com/cursorless-dev/cursorless/issues/2128#issuecomment-1856147061, we've started leaning towards requiring iteration scope between layers of any nested scope, so that would argue for having an iteration scope here cc/ @AndreasArvidsson 

3) is an interesting idea, and certainly I could see this being a fine-grained scope that gets rolled up into arg, tho we don't really support that yet

Reply:
For now I just removed the iteration scope, but keep the interpolations as args. I'll mark this unresolved in case there is some other type of iteration scope I should add in light of 2).